src/core/data-structures/observable.js

```diff
@@ -58,7 +58,7 @@ export class Observable {
     if (!this.handlers_) {
       return;
     }
-    for (const handler of this.handlers_) {
+    for (const handler of [...this.handlers_]) {
```

Comment:
We couldn't call `remove` on the handler in the middle of this operation because removing the handler would skip the next element. By duplicating the array before calling the handlers we ensure all the handlers get called even if some of them get removed.

Reply:
Great to get your feedback on this. I thought of this but doing reverse iteration could impact other parts of the code in unintended ways, are you sure we could do that without breaking anything?

Otherwise running the cleanup post-iteration makes the most sense IMO, with a bit of extra complexity.